docs/user/troubleshooting.md

```diff
@@ -165,4 +165,53 @@ You can either increase the capacity of your backend or reduce the volume of log
+
+## EOF Error When Exporting OTLP Data
+
+### Symptom
+
+The Telemetry gateway logs show errors like: `OTLP export failed: ... rpc error: code = Unavailable desc = error reading from server: EOF`.
+
+### Cause
+
+An "End of File" (EOF) error indicates that the connection to your OTLP backend was closed unexpectedly by the server. This can be caused by several issues:
+* **Incorrect Endpoint Address**: The gateway is connecting to the wrong host or port.
+* **TLS Configuration Mismatch**: The gateway's TLS settings (`insecure`, `ca`, `cert`, `key`) do not match what the backend server expects. For example, you are connecting without TLS (`insecure: true`) to a server that requires it.
+* **Network Policy**: A Kubernetes `NetworkPolicy` is blocking egress traffic from the `telemetry-operator-gateway` pods to the OTLP endpoint.
+* **Backend Unavailability**: The backend service is crashing, restarting, or is otherwise unhealthy, causing it to drop active connections.
+
+### Solution
+
+1. Check the `endpoint` value in your pipeline's `output.otlp` section and ensure the host and port are correct.
+2. Verify the TLS configuration:
+   * If your backend needs TLS, ensure **insecure** is not set to `true`.
+   * If your backend uses a custom Certificate Authority (CA), ensure the `ca` reference in your `tls` configuration is correct.
+   * If your backend requires mutual TLS (mTLS), verify that your `cert` and `key` references are correct and the secrets exist.
+3. Look for `NetworkPolicy` resources in the `kyma-system` namespace that could be blocking egress traffic from the gateway pods. You may need to create a policy to explicitly allow this traffic.
+4. Ensure the OTLP backend service is running and accessible from within the cluster. You can test connectivity from the gateway pod using tools like `curl` (for HTTP) or `grpcurl` (for gRPC).
+
+## Transform or Filter Rules Not Working
+
+### Symptom
+
+You have configured a `transform` or `filter` section in your pipeline, but the data arriving at your backend is not modified, or data you expect to be dropped is still present.
+
+### Cause
+
+This usually happens for one of the following reasons:
+* **Incorrect Execution Order**: You're filtering based on an attribute's original value, but a transformation has already changed it. Transform rules always run before filter rules.
+* **Incorrect Context Path**: Your OTTL expression references an attribute without the required explicit context path (for example, using `attributes[...]` instead of `resource.attributes[...]`).
+* **Condition Never Met**: The condition in your rule is valid but never finds a match in the data. This is often due to a case-sensitive value mismatch or a flawed regular expression.
+
+> [!NOTE]
+> If the OTTL syntax is invalid, the pipeline configuration is not generated. The corresponding error message appears in the pipeline status under the `ConfigurationGenerated` condition with reason `OTTLSpecInvalid`.
```

Comment:
If the OTTL syntax is invalid, is then the symptom that the transform/filter rule doesn't work? (the headline of this section)
Or would invalid OTTL syntax lead to another symptom? (such as: no pipeline exists (?), pipeline status "OTTLSpecInvalid").

=> Does this error truly belong under this headline?

Reply:
Decision: [Include it in the sentence above](https://github.com/kyma-project/telemetry-manager/pull/2653#discussion_r2499974376) + drop this note.